### DISCUSSION TITLE
(Typescript) Interaction with recursive items

### DISCUSSION THREAD

**Discussion by Doppelgr** [community/community#14086] [Copilot Conversations]
I just want to contribute by giving my feedback. At first i will give some context, but to resume, the issue is that it has some problems when recursive objects are involved (i manage them this way because i use mongodb to store the collections).

I have a TS Rect project. In my project i have a interface defined as
<img width="339" alt="image" src="https://user-images.githubusercontent.com/101216601/161373410-d64caf04-c053-42d9-bb79-6b663e0a4d89.png">

The interface SFolderRecursiveDTO has an array of SFolderRecursiveDTO wich can be empty.

At the start, Copilot had some troubles when this interface was envolved, for example, i tried to make a mapper from SFolderRecursiveDTO  -> BMM_FolderExplorerStorage

<img width="286" alt="image" src="https://user-images.githubusercontent.com/101216601/161373605-99d5fde0-c913-475c-a0c9-6b9428071372.png">

A ( work in progress ) interface that i use to draw the html. 

At first, it was unable to end any suggestion, because it started to assing not existing props inside BMM_FolderExplorerStorage
Like parentSFolderPropsDTO, parentParentSFolderPropsDTO, parentParentParentSFolderPropsDTO, etc. (I am sorry i didnt take screenshoots, in this moment i wasnt thinking in to share this)

When i saw this, i tried to give it some context, like

// map SFolderRecursiveDTO into BMM_FolderExplorerStorage
// map inner SFolderRecursiveDTO to BMM_Folder

After accepting some suggestions, fix them, and doing it again. i got a functional mapper that i am pleased with.

<img width="686" alt="image" src="https://user-images.githubusercontent.com/101216601/161374196-0d4a428d-86ce-4174-b955-1f6d043727d4.png">

( Copilot generated the mapFolderRecursiveDTOsToBMM_Folders but didnt use it on mapFolderRecursiveDTO, i added it manually )

I would like to point out is that it still has some problems with the recursive interface
![image](https://user-images.githubusercontent.com/101216601/161374462-37cf35b5-d1aa-433e-b1eb-30a4554cfcef.png)
<img width="817" alt="image" src="https://user-images.githubusercontent.com/101216601/161374517-40934a66-5b07-4d3c-b65e-09a8dfcbf2c9.png">
same with
// log folders prop inside testData one by one

Despite this, now most the time is working fine, but it may because it has a lot of context of my project, and may repeat its initial mistakes when a different recursive interface is involved in the project of another person. so i wanted to give my feedback.

**Comment by Marychoffel22** [comment]
Assigned

**Reply by Marychoffel22** [reply]
Pull requests

**Comment by Marychoffel22** [comment]
> I just want to contribute by giving my feedback. At first i will give some context, but to resume, the issue is that it has some problems when recursive objects are involved (i manage them this way because i use mongodb to store the collections).
> 
> 
> 
> I have a TS Rect project. In my project i have a interface defined as
> 
> <img width="339" alt="image" src="https://user-images.githubusercontent.com/101216601/161373410-d64caf04-c053-42d9-bb79-6b663e0a4d89.png">
> 
> 
> 
> The interface SFolderRecursiveDTO has an array of SFolderRecursiveDTO wich can be empty.
> 
> 
> 
> At the start, Copilot had some troubles when this interface was envolved, for example, i tried to make a mapper from SFolderRecursiveDTO  -> BMM_FolderExplorerStorage
> 
> 
> 
> <img width="286" alt="image" src="https://user-images.githubusercontent.com/101216601/161373605-99d5fde0-c913-475c-a0c9-6b9428071372.png">
> 
> 
> 
> A ( work in progress ) interface that i use to draw the html. 
> 
> 
> 
> At first, it was unable to end any suggestion, because it started to assing not existing props inside BMM_FolderExplorerStorage
> 
> Like parentSFolderPropsDTO, parentParentSFolderPropsDTO, parentParentParentSFolderPropsDTO, etc. (I am sorry i didnt take screenshoots, in this moment i wasnt thinking in to share this)
> 
> 
> 
> When i saw this, i tried to give it some context, like
> 
> 
> 
> // map SFolderRecursiveDTO into BMM_FolderExplorerStorage
> 
> // map inner SFolderRecursiveDTO to BMM_Folder
> 
> 
> 
> After accepting some suggestions, fix them, and doing it again. i got a functional mapper that i am pleased with.
> 
> 
> 
> <img width="686" alt="image" src="https://user-images.githubusercontent.com/101216601/161374196-0d4a428d-86ce-4174-b955-1f6d043727d4.png">
> 
> 
> 
> ( Copilot generated the mapFolderRecursiveDTOsToBMM_Folders but didnt use it on mapFolderRecursiveDTO, i added it manually )
> 
> 
> 
> I would like to point out is that it still has some problems with the recursive interface
> 
> ![image](https://user-images.githubusercontent.com/101216601/161374462-37cf35b5-d1aa-433e-b1eb-30a4554cfcef.png)
> 
> <img width="817" alt="image" src="https://user-images.githubusercontent.com/101216601/161374517-40934a66-5b07-4d3c-b65e-09a8dfcbf2c9.png">
> 
> same with
> 
> // log folders prop inside testData one by one
> 
> 
> 
> Despite this, now most the time is working fine, but it may because it has a lot of context of my project, and may repeat its initial mistakes when a different recursive interface is involved in the project of another person. so i wanted to give my feedback.
> 
>
